RuriLib/Blocks/Interop/Methods.cs

```diff
@@ -40,14 +41,57 @@ public static string ShellCommand(BotData data, string executable, string argume
          * of writing C# code that calls these methods where necessary once it's transpiled.
          */
 
-        public static async Task<T> InvokeNode<T>(BotData data, string scriptFile, object[] parameters)
+        public static async Task<T> InvokeNode<T>(BotData data, (string, string) scriptOrFile, object[] parameters, bool isScript = false)
         {
             data.Logger.LogHeader();
-            var result = await StaticNodeJSService.InvokeFromFileAsync<T>(scriptFile, null, parameters, data.CancellationToken).ConfigureAwait(false);
+            T result;
+
+            if (isScript)
+            {
+                result = await InvokeFromStringOrCache<T>(data, scriptOrFile, parameters);
+            }
+            else
+            {
+               
+                result = await InvokeFromFile<T>(data, scriptOrFile.Item2 /*hash*/, parameters);
```

Comment:
This has to be changed anyways, according to the comment above, but I suggest that you use named tuples in the future instead of standard tuples. `Item1` and `Item2` are not good names.